### ISSUE TITLE
Will all of the model weights be saved?

### ISSUE THREAD

**Issue by function2-llx** [zjohn77/lightning-mlflow-hf#1]
Hello, thank you so much for sharing this repository! I want to ask a question about the implementation.

According to the code, the `ModelCheckpoint` callback is used to save the model weights. However, according to my understanding, it will save the whole model weights, while usually only the adapter weights is saved for memory efficiency. Could you please share your thoughts on this?

https://github.com/zjohn77/lightning-mlflow-hf/blob/cf1b6b98211473b15019e7e90b6a4acaa1a2b480/lightning_mlflow/train.py#L57-L76

**Comment by zjohn77** [comment]
Yes, that's a good observation. This checkpoint callback from PyTorch Lightning indeed saves all weights, including the frozen weights, when it would have been more efficient just checkpointing the fine-tuned weights. 

Also, I think that this efficiency issue would exist not only in LoRA fine-tuning but also in linear probing, or for that matter any kind of fine-tuning. The checkpoint callback currently does not check for or handle what type of fine-tuning is happening. Maybe it should though (this could be a ticket for the PyTorch Lightning project). Do other checkpointing tools (for the base PyTorch, for the Hugging Face Trainer, or for Tensorflow) have such a functionality?

**Comment by function2-llx** [comment]
FYI, I found an related issue in the PyTorch Lightning repository, where it is suggested that one may manually filter the weights to be saved by overriding the `state_dict` method of the `LightningModule`.
https://github.com/Lightning-AI/pytorch-lightning/issues/19149#issuecomment-1853282628

Also, I believe the official way for saving weights of adapters from the peft library is to use the [`PeftModel.save_pretrained`](https://huggingface.co/docs/peft/en/package_reference/peft_model#peft.PeftModel.save_pretrained) method.
